libbeat/esleg/eslegclient/connection.go

```diff
@@ -311,16 +314,113 @@ func (conn *Connection) Close() error {
 	return nil
 }
 
+type httpClientProxySettings struct {
+	*httpcommon.HTTPClientProxySettings
+}
+
+// ProxyDialer is a dialer that can be registered to golang.org/x/net/proxy
+func (settings *httpClientProxySettings) ProxyDialer(_ *url.URL, forward proxy.Dialer) (proxy.Dialer, error) {
```

Comment:
By convention method receivers in Go are named after the first letter of the type:
```suggestion
func (h *httpClientProxySettings) ProxyDialer(_ *url.URL, forward proxy.Dialer) (proxy.Dialer, error) {
```

Reply:
Thanks! Shorten the receiver name accordingly.